sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala

```diff
@@ -273,6 +274,24 @@ class CatalystTypeConvertersSuite extends SparkFunSuite with SQLHelper {
     assert(errMsg.contains("long overflow"))
   }
 
+  test("SPARK-35726: Truncate java.time.Duration by fields of day-time interval type") {
+    val duration = Duration.ofSeconds(90061)
```

Comment:
Could you test a Duration with nanos part.

Reply:
done